UI/MapboxNavigationUI/ArrowStyleLayer.swift

```diff
@@ -0,0 +1,101 @@
+import Foundation
+import Mapbox
+import MapboxDirections
+import MapboxNavigation
+
+class ArrowStyleLayer {
+    
+    class func add(to mapView: MGLMapView,
+                   nextStep: RouteProgress,
+                   currentManeuverArrowStrokePolylines: inout [ArrowFillPolyline],
+                   currentManeuverArrowPolylines: inout [ArrowFillPolyline]) {
```

Comment:
Since I think this method could more naturally be expressed as a category method on MGLMapView. It’s also important to ensure that argument names agree with their types. For example, `nextStep` would be the name of a RouteStep, not a RouteProgress.

```swift
extension MGLMapView {
    func addArrowStyleLayer(for routeProgress: RouteProgress)
}
```

Ideally `addArrowLayer(for:)` would be a category method on MGLStyle, but that would preclude the use of `MGLMapView.metersPerPoint(atLatitude:)`.

`inout` arguments should be a last resort. As an alternative, consider filtering MGLMapView’s built-in `annotations` property by class. If that approach isn’t satisfactory and it really is necessary to store a list of polylines independently of the `annotations` property, then that’s a sign that we need to provide a subclass of MGLMapView (RouteMapView?).